src/components/StakingVaults/EarnOpportunities.tsx

```diff
@@ -85,21 +85,22 @@ export const EarnOpportunities = ({ assetId, accountId }: EarnOpportunitiesProps
         return navigate('/rfox')
       }
 
-      navigate(location.pathname, {
-        state: {
-          search: qs.stringify({
-            chainId,
-            contractAddress,
-            assetNamespace,
-            assetReference,
-            highestBalanceAccountAddress: opportunity.highestBalanceAccountAddress,
-            rewardId: rewardAddress,
-            provider,
-            type,
-            modal: 'overview',
-          }),
-          state: { background: location },
-        },
+      // @ts-ignore that's incorrect according to types but is absolutely valid
+      // The correct signature doesn't cut it and will bork DeFi row click in account/asset page
+      navigate({
+        pathname: location.pathname,
```

Comment:
FWIW https://ultimatecourses.com/blog/navigate-to-url-query-strings-search-params-react-router

For all intents and purposes, this is correct, think react-router may be a bit derp typings-wise when mixing all 3 of pathname, search and state (or perhaps it is when not honouring the `[...args]` signature where args is `[pathname, NavigateOptions]`, not sure, but either way this works, and I couldn't get this to work with the regular signature)